### DISCUSSION TITLE
OrderBy relationated field

### DISCUSSION THREAD

**Discussion by rodolphonetto** [adonisjs/core#1537] [Help] [answered]
Hi guys, I can't use a relationated field to orderBy a query, let me show you an exampe

I have that

```
    query
      .with('solicitation_type')
      .with('category')
      .with('status')
      .with('states')
      .with('cities')
      .with('admin_status_history', (builder) => {
        builder.orderBy('created_at', 'desc')
      })
      .orderBy('solicitation_type.title', 'desc')
```

It gives me that error Unknown column 'solicitation_types.title' in 'order clause'

How can I order by a field in the relation?

**Comment by thetutlage** [comment, marked as answer]
Yes coz relationships are loaded in a separate query after the main query. What you need is a order by subquery. Following should do it.

```js
.orderByRaw(
  Database.raw('select title from solicitation_types where solicitation_types.FK = main_table.PK'),
  'desc'
)
```

**Reply by rodolphonetto** [reply]
Thanks big boss, it worked perfectly

**Reply by DZ8540** [reply]
> Yes coz relationships are loaded in a separate query after the main query. What you need is a order by subquery. Following should do it.
> 
> ```js
> .orderByRaw(
>   Database.raw('select title from solicitation_types where solicitation_types.FK = main_table.PK'),
>   'desc'
> )
> ```

Hey big boss, can you show me how i can make this at lucid 16.3.2 please?

I have a services and users tables, and i need to order services by users' rating column.
